plugin/src/errors.ts

```diff
@@ -13,3 +16,13 @@ export class OperationError extends Error {
     this.node = node;
   }
 }
+
+export class HttpError extends Error {
+  public response: Response;
+
+  constructor(response: Response) {
+    super(response.statusText);
+    Object.setPrototypeOf(this, HttpError.prototype);
+    this.response = response;
+  }
```

Comment:
I'm curious where you find this piece of code to do errors.
When using classes prototype is already getting inherited, so you don't need `setPrototypeOf`.

The correct code to actually get correct stacktraces and such should be:
```js
  constructor(response: Response) {
    super(response.statusText);

    this.name = `HttpError`;
    this.response = response;

    Error.captureStackTrace(this, HttpError)
  }
```

We'll set the name to `HttpError` else it would still be named `Error`.

This explains why captureStackTrace is useful: https://nodejs.org/api/errors.html#errors_error_capturestacktrace_targetobject_constructoropt

Reply:
This was in the TypeScript docs for subclasses the Error class. The reason is that insranceof does not work correctly without setting prototype. 

I don't know if it changes or not but this code is tested and works. I'm off this week as well but if you are still certain there is a better way to do this I will happily revisit on my return.

Reply:
Enjoy your vacation :)
https://github.com/gatsbyjs/gatsby-source-shopify-experimental/pull/111